app.js

```diff
@@ -60,9 +61,15 @@ app.use(expressValidator({
     }
 }));
 
-/**
- * Passport-middleware
- */
+app.use(passport.initialize());
+app.use(passport.session());
+
+app.set("view engine", "ejs");
+
+var user = require('./models/users');
+passport.use(new localStrategy(user.authenticate()));
+passport.serializeUser(user.serializeUser());
+passport.deserializeUser(user.deserializeUser());
 
 app.listen(port, () => {
     console.log("Server started at port " + port);
```

Comment:
@vsvipul You forgot to maintain a global variable for the user. That's why the session was not working well.
Before `app.use("/", publicRoute);` Line77 add:
`app.get("*", (req, res, next) => {
    res.locals.user = req.user || null;
    console.log("User: " + res.locals.user);
    next();
})`

Reply:
i have passed local user for every res.render request instead of declaring a global variable for the user.